src/main/webapp/confirmation.jsp

```diff
@@ -0,0 +1,38 @@
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Confirmation Page</title>
+</head>
+<body>
+<jsp:include page="userInformations.jsp" />
+
+<h4>Subscription</h4>
+<c:set var="user" value="${sessionScope.user}"/>
+<c:set var="subscription" value="${sessionScope.subscription}"/>
+<c:set var="servicePackage" value="${subscription.getServicePackage()}"/>
```

Comment:
if the user is on this page and waits long enough for the session to expire then the functionality will not work anymore, and we will have a 500 error on the page for trying to access an getter on a null object. the workaround for this would be to put the subscription on the request when redirecting to this page.
this is a discussion: are we ok with this or should this be refactored?

Reply:
In case the session is invalid and you try to reload the page you're getting a bad request error (because the subscription is null so you end up in the if inside the doGet for the page confirmation)